x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/RestrictedTrustManagerTests.java

```diff
@@ -92,6 +97,32 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 
         numberOfClusters = scaledRandomIntBetween(2, 8);
         numberOfNodes = scaledRandomIntBetween(2, 8);
+        fields = randomNonEmptySubsetOf(Set.of(SAN_OTHER_COMMON, SAN_DNS));
+    }
+
+    public void testTrustsOnlyNameDns() throws Exception {
+        final Path cert = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode_updated.crt");
+        baseTrustManager = CertParsingUtils.getTrustManagerFromPEM(List.of(cert));
+        X509Certificate[] certs = CertParsingUtils.readX509Certificates(Collections.singletonList(cert));
+        assertTrue(certs[0].getSubjectAlternativeNames().stream().filter(pair -> (Integer) pair.get(0) == 0).findAny().isEmpty());
+        certificates.put("onlyDns", certs);
+        List<String> validDnsNames = randomNonEmptySubsetOf(
+            List.of("localhost", "localhost.localdomain", "localhost4", "localhost4.localdomain4", "localhost6", "localhost6.localdomain6")
+        );
+        final CertificateTrustRestrictions restrictions = new CertificateTrustRestrictions(validDnsNames);
+        final RestrictedTrustManager trustManager = new RestrictedTrustManager(baseTrustManager, restrictions, Set.of(SAN_DNS));
+        assertTrusted(trustManager, "onlyDns");
+    }
+
+    public void testTrustsOnlyNameOther() throws Exception {
+        final Path cert = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/nodes/trusted.crt");
+        baseTrustManager = CertParsingUtils.getTrustManagerFromPEM(List.of(cert));
+        X509Certificate[] certs = CertParsingUtils.readX509Certificates(Collections.singletonList(cert));
+        assertTrue(certs[0].getSubjectAlternativeNames().stream().filter(pair -> (Integer) pair.get(0) == 2).findAny().isEmpty());
+        certificates.put("onlyOtherName", certs);
+        final CertificateTrustRestrictions restrictions = new CertificateTrustRestrictions(List.of("node.trusted"));
+        final RestrictedTrustManager trustManager = new RestrictedTrustManager(baseTrustManager, restrictions, Set.of(SAN_OTHER_COMMON));
+        assertTrusted(trustManager, "onlyOtherName");
     }
```

Comment:
I don't think we have any tests that verify that (for example) the `SAN_DNS` names aren't used if the `field` is set to `SAN_OTHER_COMMON` (the default.

That is, with a cert that was generated from:
```
  - ip: 50.100.150.200
  - dns: search.example.com
  - cn: instance03.cluster02.elasticsearch
```

We should accept the cert with either:
- `trust.subject_name=*.cluster02.elasticsearch`, `x509_fields=[ "subjectAltName.otherName.commonName" ]` (that is, the default value for `x509_fields`)
- `trust.subject_name=*.example.com`, `x509_fields=[ "subjectAltName.dnsName" ]`

But none of these:
- `trust.subject_name=*.cluster02.elasticsearch`, `x509_fields=[ "subjectAltName.dnsName" ]`
- `trust.subject_name=*.example.com`, `x509_fields=[ "subjectAltName.otherName.commonName" ]`
- `trust.subject_name=*.150.200`, `x509_fields=[ "subjectAltName.otherName.commonName", "subjectAltName.dnsName" ]`

I can see that the code will work correctly in these cases, but I can't see tests for it.

Reply:
The randomly picked x509_fields [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/RestrictedTrustManagerTests.java#L100) should provide coverage that `SAN_DNS` names aren't used if the field is set to `SAN_OTHER_COMMON` (and vice-versa). Since values are identical on the certificate, there is a chance that we could be reading from the wrong value if something else is buggy (but the other tests should cover that scenario). However it's pretty easy to add an explicit test for this case and will do so on a follow up PR.